ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -643,7 +643,7 @@ public void setMaxLength(ReactEditText view, @Nullable Integer maxLength) {
         if (!replaced) {
           newFilters = new InputFilter[currentFilters.length + 1];
           System.arraycopy(currentFilters, 0, newFilters, 0, currentFilters.length);
-          currentFilters[currentFilters.length] = new InputFilter.LengthFilter(maxLength);
+          newFilters[currentFilters.length] = new InputFilter.LengthFilter(maxLength);
```

Comment:
@WonShaw How do you trigger the error?

>ReactTextInputManager.setMaxLength will throw ArrayIndexOutOfBoundsException when inserting LengthFilter to view's non-empty mFilters


quick analysis of the code, seems to me that the runtime is triggered by replaced being false because `currentFilters[i]` is not  `instanceof InputFilter.LengthFilter`

https://github.com/facebook/react-native/blob/a9bc3853f0cf2bb32999335c0aaf90194ea9084d/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java#L638-L641

https://developer.android.com/reference/android/text/InputFilter.LengthFilter#LengthFilter(int)

https://github.com/facebook/react-native/blob/a9bc3853f0cf2bb32999335c0aaf90194ea9084d/Libraries/Components/TextInput/TextInput.js#L531

I'm trying to guess your use case to trigger this error. Thanks a lot :pray: sorry for disturbing you

Reply:
It's just a logic error. Cannot trigger it just by using react native, because react native won't insert other InputFilter except LengthFilter.
But if you expose your custom EditText by extending ReactTextInputManager and ReactEditText, it may be triggered.

Reply:
In short, should append the LengthFilter to newFilters rather than currentFilters.

Reply:
@WonShaw Thanks a lot for the explanation

would it be possible to override this method in your child class inheriting from ReactTextInputManager or it will not work?

```java
CustomTextInputManager extends ReactTextInputManager {
  public void setMaxLength(ReactEditText view, @Nullable Integer maxLength) {
      super.setMaxLength()
      // your logic
  }
}
```
Sorry If my statement is incorrect or imprecise, I don't have lots of experience in ReactNative [NativeModules](https://reactnative.dev/docs/0.62/native-components-android).

I believe your solution is very good. Thanks a lot for contributing to ReactNative :pray: :peace_symbol: :heart: 
I agree with your request for change. :pray:

Reply:
Actually, *override this method in your child class inheriting from ReactTextInputManager* that’s my solution to this problem because I cannot change the react native aar in our project.

Reply:
All checks have passed. Thank you~

Reply:
@WonShaw AWESOME. Thanks a lot for the incredible work. :pray: :peace_symbol: